subprojects/core/src/main/java/org/gradle/api/internal/plugins/DefaultPluginManager.java

```diff
@@ -102,19 +107,28 @@ public PluginContainer getPluginContainer() {
 
     @Override
     public void apply(PluginImplementation<?> plugin) {
-        doApply(plugin);
+        doApplyThroughBuildOperation(plugin);
     }
 
     public void apply(String pluginId) {
         PluginImplementation<?> plugin = pluginRegistry.lookup(DefaultPluginId.unvalidated(pluginId));
         if (plugin == null) {
             throw new UnknownPluginException("Plugin with id '" + pluginId + "' not found.");
         }
-        doApply(plugin);
+        doApplyThroughBuildOperation(plugin);
     }
 
     public void apply(Class<?> type) {
-        doApply(pluginRegistry.inspect(type));
+        doApplyThroughBuildOperation(pluginRegistry.inspect(type));
+    }
+
+    private void doApplyThroughBuildOperation(final PluginImplementation<?> plugin) {
+        new ApplyPluginBuildOperation(plugin, buildOperationExecutor).run(new Action<BuildOperationContext>() {
+            @Override
+            public void execute(BuildOperationContext buildOperationContext) {
+                doApply(plugin);
```

Comment:
Changed in https://github.com/gradle/gradle/pull/1033/commits/de275793dee72a0b9ed2f122b7eb13fccee78c5c

it is now wrapping everything in the `if (adder != null)` block from inside `doAplly()`.